addons/mail/controllers/main.py

```diff
@@ -112,7 +112,12 @@ def receive(self, req):
     def read_followers(self, follower_ids):
         result = []
         is_editable = request.env.user.has_group('base.group_no_one')
-        for follower in request.env['mail.followers'].browse(follower_ids):
+        follower_recs = request.env['mail.followers'].sudo().browse(follower_ids)
+        res_ids = follower_recs.mapped('res_id')
+        if follower_recs:
+            res_model = follower_recs[0].res_model
```

Comment:
We should be careful that here (even if it will be the case from the interface) we are not sure that the 'res_model' is the same for all the followers. If someone is calling the method with a RPC call for example, we could authorize the modifications if the user has the 'write' access rule on the record linked to the first follower.

Reply:
I'm gonna test your dev and make the modifications myself ! ;)